lib/Exception/UserTokenException.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @author Sujith Haridasan <sharidasan@owncloud.com>
+ *
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\UserManagement\Exception;
+
+class UserTokenException extends \Exception {
+	const TOKEN_AND_TIME_INVALID = 10;
+	const TOKEN_TIME_EXPIRED = 20;
+	const TOKEN_MISMACTH = 30;
+
+	public function __construct($exceptionMessageType, \Exception $previous = null) {
+		$message = '';
+		$code = $exceptionMessageType;
+		if ($exceptionMessageType === self::TOKEN_AND_TIME_INVALID) {
```

Comment:
Not sure if this is a common pattern in the PHP world. As far as I know we never used this exception pattern before so I'd prefer we use existing patterns instead of introducing new ones. See my comment here https://github.com/owncloud/user_management/pull/48/files#r212546994

also the message says "could not set password" but the exception is called UserTokenException instead of CouldNotSetPasswordException which feels inconsistent.

Reply:
Let me know if this looks ok https://github.com/owncloud/user_management/pull/48/files#r212614274.